VAULT-42829: Create a new billing config endpoint to set custom retention months (#2494)
hashicorp/web-unified-docs: content/vault/v2.x/content/api-docs/system/billing/config.mdx

---
layout: api
page_title: /sys/billing/config - HTTP API
description: >-
  The `/sys/billing/config` endpoint manages billing data retention configuration.
# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2026-05-21T14:00:00-05:00
last_modified: 2026-05-21T14:00:00-05:00
# END AUTO GENERATED METADATA
---

# `/sys/billing/config`

@include 'alerts/restricted-root.mdx'

> The `/sys/billing/config` endpoint allows you to configure how long Vault retains billing data.

Reviewer: I'm assuming the retention period only applies to `sys/billing/overview` data? Or does it extend to certificate counts (`sys/billing/certificates`) and client counts?

```suggestion
The `/sys/billing/config` endpoint allows you to configure how long Vault retains `/sys/billing/overview` data.
```

Author: Good point, yes! it only applies to sys/billing/overview endpoint

> By default, Vault retains billing data for 37 months, including the current month. You can configure the retention period to be between 13 months (minimum) and 72 months (maximum).

Reviewer: ```suggestion
By default, Vault retains `sys/billing/overview` data for 37 months, including the current month. You can configure the retention period to be between 13 months (minimum) and 72 months (maximum).
```

> Increasing the retention period cannot restore historical data for months that was previously outside the retention period, but will retain data from the previous retention period.

Reviewer: Don't love this phrasing. @schavis Do you have any suggestions?

We're trying to say here that:
- If the customer was retaining data for 13 months, changing this to 37 months won't instantly make 37 months of data show up.
- As time goes by, after the user increases retention period, Vault will accumulate up to 37 months of data.

Reviewer 2: I added a suggestion below. Let me know what you think

> Increasing the retention period cannot restore historical data for months that was previously outside the retention period, but will retain data from the previous retention period.

Reviewer 2: ```suggestion
Vault does not restore historical data when you increase the retention period.
Instead, Vault accumulates new data up to the new retention period before
advancing the retention window. For example, if you increase your retention
window from 13 months to 37 months, Vault does not advance the retention window
until it accumulates 37 months of data.
```

Author: I like it, @gsantos-hc if you like this too, we can commit her suggestion:)
